addons/base_vat/base_vat.py

```diff
@@ -328,3 +334,33 @@ def check_vat_tr(self, vat):
             return int(vat[9]) == c1 and int(vat[10]) == c2
 
         return False
+
+    def get_address(self, cr, uid, vat, context=None):
+        vals = {}
+        user_company = self.pool['res.users'].browse(cr, uid, uid, context=context).company_id
+        if user_company.vat_check_vies:
+            try:
+                vat_country, vat_number = self._split_vat(vat)
+                client = Client(vatnumber.VIES_URL)
+                response = client.service.checkVat(countryCode=vat_country, vatNumber=vat_number)
+                if response['valid']:
+                    country_id = self.pool['res.country'].search(cr, uid, [('code', '=ilike', vat_country)], context=context)
+                    address = str(response['address'].encode('utf-8')).split('\n')
+                    vals.update(street=address[0], street2=address[1] if len(address) > 1 else False, country_id=country_id[0] if country_id else False)
+            except:
+                # see http://ec.europa.eu/taxation_customs/vies/checkVatService.wsdl
+                # Fault code may contain INVALID_INPUT, SERVICE_UNAVAILABLE, MS_UNAVAILABLE,
+                # TIMEOUT or SERVER_BUSY. There is no way we can get any address information
+                # from VIES if any of these arise, so we simply bypass setting address data.
+                pass
+        return vals
+
+    def create(self, cr, uid, vals, context=None):
+        if vals.get('vat'):
+            vals.update(self.get_address(cr, uid, vals['vat'], context=context))
+        return super(res_partner, self).create(cr, uid, vals, context=context)
+
+    def write(self, cr, uid, ids, vals, context=None):
+        if vals.get('vat'):
+            vals.update(self.get_address(cr, uid, vals['vat'], context=context))
+        return super(res_partner, self).write(cr, uid, ids, vals, context=context)
```

Comment:
@rim-odoo,
Do you think it's good to call VIES Service with every onchange ?
Regarding Write if user performs write in batch  it means user want's to set same values for all ids then if vat number is found in 'vals' then it is meant to be written to all ids. isn't it ?